### PR TITLE
Add tbconnect app

### PR DESCRIPTION
**Pull request by erikh360**

### CODE REVIEW

File: tbconnect/admin.py [hunk 1]
```diff
@@ -0,0 +1,3 @@
+from django.contrib import admin
+
+# Register your models here.
```
Comment (rudigiesler) [inline tbconnect/admin.py]: We probably do want a django admin, helps a lot for QA

File: tbconnect/tests.py [hunk 2]
```diff
@@ -0,0 +1,3 @@
+from django.test import TestCase
+
+# Create your tests here.
```
Comment (rudigiesler) [inline tbconnect/tests.py]: We can probably remove this file, and instead rely on the separate `test_views.py` etc

File: tbconnect/views.py [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+from rest_framework.viewsets import GenericViewSet
+
+from .models import TBCheck
+from .serializers import TBCheckSerializer
+
+
+class TBCheckViewSet(GenericViewSet, CreateModelMixin):
+    queryset = TBCheck.objects.all()
+    serializer_class = TBCheckSerializer
+    permission_classes = (DjangoModelPermissions,)
```
Comment (rudigiesler) [inline tbconnect/views.py:12]: Do we not want the `created_by` field filled in according to the user that made the request?

File: tbconnect/models.py [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+    GENDER_OTHER = "other"
+    GENDER_NOT_SAY = "not_say"
+    GENDER_CHOICES = (
+        (GENDER_MALE, "Male"),
+        (GENDER_FEMALE, "Female"),
+        (GENDER_OTHER, "Other"),
+        (GENDER_NOT_SAY, "Rather not say"),
+    )
+
+    id = models.UUIDField(primary_key=True, default=uuid.uuid4, editable=False)
```
Comment (rudigiesler) [inline tbconnect/models.py]: I actually consider this a mistake on the original model. Pagination is much easier on normal auto incrementing primary keys, timestamps can have overlaps. But happy with it either way.

File: tbconnect/models.py [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+    source = models.CharField(max_length=255)
+    province = models.CharField(max_length=6, choices=PROVINCE_CHOICES)
+    city = models.CharField(max_length=255)
+    age = models.CharField(max_length=5, choices=AGE_CHOICES)
+    gender = models.CharField(
+        max_length=7, choices=GENDER_CHOICES, blank=True, default=""
+    )
+    location = models.CharField(
+        max_length=255, blank=True, default="", validators=[geographic_coordinate]
+    )
```
Comment (rudigiesler) [inline tbconnect/models.py]: Just checking that we mean to allow `gender` and `location` to be optional, I'm not sure what the user flow is here.

File: tbconnect/models.py [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+    cough = models.CharField(max_length=13, choices=COUGH_CHOICES)
+    fever = models.BooleanField()
+    sweat = models.BooleanField()
+    weight = models.BooleanField()
+    exposure = models.CharField(max_length=9, choices=EXPOSURE_CHOICES)
+    tracing = models.BooleanField(help_text="Whether the NDoH can contact the user")
```
Comment (rudigiesler) [inline tbconnect/models.py:80]: Do we maybe want a timestamp of when it was completed?

Also an `external_id` field that allows for deduplication is helpful for retrying failed requests.

File: tbconnect/views.py (hunk 3 above)
Comment (erikh360) [reply tbconnect/views.py:12]: This happens in the `BaseEventSerializer` and I inherit from it here: https://github.com/praekeltfoundation/healthcheck/pull/12/files#diff-ce4827534265fe613e25b92fea0631efR6

Comment (rudigiesler) [reply tbconnect/views.py:12]: 👍  missed that

**Review by rudigiesler — APPROVED** [review]

**Pull request merged**